test/testAdaptors/CompoundV2AdditionalTests.t.sol

```diff
@@ -0,0 +1,1551 @@
+
+        // compare method results
+        uint256 relativeDiff;
+        if (healthFactorOptionA >= healthFactorOptionB) {
+            relativeDiff = ((healthFactorOptionA - healthFactorOptionB) * 1e18) / healthFactorOptionA;
+        } else {
+            relativeDiff = ((healthFactorOptionB - healthFactorOptionA) * 1e18) / healthFactorOptionB;
+        }
+
+        assertGt(1e16, relativeDiff, "relativeDiff cannot exceed 1bps."); // ensure that relativeDiff !> 1bps (1e16)
+    }
+
+    // add collateral
+    // try borrowing from same market
+    // a borrow position will open up in the same market; cellar has a cToken position from lending underlying (DAI), and a borrow balance from borrowing DAI.
+    // test borrowing going up to the HF, have it revert because of HF. 
+    // test redeeming going up to the HF, have it revert because of HF.
+    function testBorrowInSameCollateralMarket() external {
+        uint256 initialAssets = cellar.totalAssets();
+        uint256 assets = 1e18;
+        // assets = bound(assets, 0.1e18, 1_000_000e18);
+        deal(address(DAI), address(this), assets);
+        cellar.deposit(assets, address(this)); // holding position is cDAI (w/o entering market)
+
+        // enter market
+        Cellar.AdaptorCall[] memory data = new Cellar.AdaptorCall[](1);
+        bytes[] memory adaptorCalls = new bytes[](1);
+        {
+            adaptorCalls[0] = _createBytesDataToEnterMarketWithCompoundV2(cDAI);
+            data[0] = Cellar.AdaptorCall({ adaptor: address(cTokenAdaptor), callData: adaptorCalls });
+        }
+        cellar.callOnAdaptor(data);
+        uint256 initialcDaiBalance = cDAI.balanceOf(address(cellar));
+        console.log("initialCDaiBalance: %s", initialcDaiBalance);
+
+        // borrow from same market unlike other tests
+        {
+            adaptorCalls[0] = _createBytesDataToBorrowWithCompoundV2(cDAI, (assets/2));
+            data[0] = Cellar.AdaptorCall({ adaptor: address(compoundV2DebtAdaptor), callData: adaptorCalls });
+        }
+        cellar.callOnAdaptor(data);
+
+        // call should go through, and it records a borrow balance in cToken market.
+        assertEq(
+            DAI.balanceOf(address(cellar)),
+            assets / 2,
+            "Borrowing from a market that cellar has lent out to already means they are just withdrawing some of their lent out initial amount."
+        );
+        assertEq(
+            cDAI.borrowBalanceStored(address(cellar)),
+            assets / 2,
+            "CompoundV2 market should show borrowed, even though cellar is also supplying said underlying asset."
+        );
+        assertEq(
+            cDAI.balanceOf(address(cellar)),
+            initialcDaiBalance,
+            "CompoundV2 market should show same amount cDai for cellar."
+        );
+
+        uint256 lowerThanMinHF = 1.05e18;
+        uint256 amountToWithdraw = _generateAmountBasedOnHFOptionA(
```

Comment:
@crispymangoes, this is what my comment is referring to in regards to calculating the proper `amountToWithdraw` that will make this test pass.